pkg/store/proxy_merge.go

```diff
@@ -814,15 +754,15 @@ func (l *eagerRespSet) At() *storepb.SeriesResponse {
 		return nil
 	}
 
-	return l.bufferedResponses[l.i]
+	return l.bufferedResponses[l.i-1]
```

Comment:
Is it because we call `Next` before `At` so we need to change to `i-1`?

Reply:
Yeah, our dedup heap tree was a bit buggy - we were calling At() before Next() was ever called. The loser tree properly calls Next first hence this change